### PR TITLE
tests: Add tests for various error flows

### PR DESCRIPTION
**Pull request by firasj**
This series adds various error flow tests on post send/receive operations, and adds instances of these tests for UD and SRD.

### CODE REVIEW

File: tests/test_efa_srd.py [hunk 1]
```diff
@@ -31,6 +31,23 @@ def create_players(self, send_ops_flags=0, qp_count=8):
         self.client.pre_run(self.server.psns, self.server.qps_num)
         self.server.pre_run(self.client.psns, self.client.qps_num)
 
+    def full_sq_bad_flow(self):
```
Comment (EdwardSro) [inline tests/test_efa_srd.py:34]: Is there a reason why you chose to skip UD in this case?

**Review by EdwardSro — CHANGES_REQUESTED** [review]

**Review by jgunthorpe — CHANGES_REQUESTED** [review]
Address Edward's remarks

**Review by firasj — COMMENTED** [review]
Sorry for the delay.
I have addressed all of your comments and updated the commits.

File: tests/utils.py [hunk 2]
```diff
@@ -1227,3 +1227,40 @@ def is_datagram_qp(agr_obj):
...
+    Check post_recive on rq while qp is in invalid state.
+    - Change qp's state to IBV_QPS_RESET
+    - Verify post receive on qp fails
+    :param traffic_obj: Traffic object, base class is TrafficResources.
+    :return: None.
+    """
+    qp_attr = QPAttr(qp_state=e.IBV_QPS_RESET, cur_qp_state=e.IBV_QPS_RTS)
+    traffic_obj.qps[0].modify(qp_attr, e.IBV_QP_STATE)
+    recv_wr = get_recv_wr(traffic_obj)
+    with unittest.TestCase().assertRaises(PyverbsRDMAError) as ex:
```
[the remark this answers is not in the document]
Comment (firasj) [reply tests/utils.py]: Replaced `traffic_obj` with `test_obj`, an instance of `RDMATestCase`.

**Review by EdwardSro — APPROVED** [review]

**Comment by firasj** [comment]
Kind reminder.

**Comment by rleon** [comment]
Thanks for the reminder

**Pull request merged**